vllm/model_executor/models/ultravox.py

```diff
@@ -85,15 +87,14 @@ def _get_feature_extractor(self) -> WhisperFeatureExtractor:
         hf_processor = self._get_hf_processor()
         return hf_processor.audio_processor.feature_extractor  # type: ignore
 
-    def _get_hf_mm_data(
+    def _to_mm_items(
         self,
-        mm_items: MultiModalDataItems,
-    ) -> tuple[dict[str, Any], dict[str, Any]]:
-        # resample audio to the model's sampling rate
+        mm_data: MultiModalDataDict,
+    ) -> MultiModalDataItems:
         feature_extractor = self._get_feature_extractor()
-        mm_items.resample_audios(feature_extractor.sampling_rate)
-
-        return super()._get_hf_mm_data(mm_items)
+        parser = MultiModalDataParser(
+            target_sr=feature_extractor.sampling_rate)
```

Comment:
Seems that the parser is always fixed during processing, perhaps we can initialize the parser in `MultimodaProcessor` initialization to avoid inheriting `_to_mm_items`?

Reply:
Sure!